app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsViewModel.kt

```diff
@@ -9,22 +11,36 @@ import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.Ad
 import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.AdministratorControlsProfileViewModel
 import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.AdministratorControlsItemViewModel
 import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.ProfileId
 import org.oppia.app.viewmodel.ObservableArrayList
 import org.oppia.app.viewmodel.ObservableViewModel
+import org.oppia.domain.profile.ProfileManagementController
+import org.oppia.util.logging.Logger
 import javax.inject.Inject
 
 /** [ViewModel] for [AdministratorControlsFragment]. */
 @FragmentScope
-class AdministratorControlsViewModel @Inject constructor() : ObservableViewModel() {
+class AdministratorControlsViewModel @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val logger: Logger,
+  private val profileManagementController: ProfileManagementController
```

Comment:
Your implementation is not working because all these parameters are never getting set in your viewmodel.
To make it correct check the implementation of `StoryChapterSummaryViewModel` inside `StoryViewModel` and how `StoryChapterSummaryViewModel` takes care of the rest. Meaning that your `AdministratorControlsViewModel` will contain the live data process everything and then create a `AdministratorControlsDownloadPermissionsViewModel` which will have all the final things which are required in xml file.

Reply:
Done. Thanks for the suggestion. Download Permissions are working and test-cases are passing as well.